From 9eac5545abf909cac99b607f9c3b41e078831f98 Mon Sep 17 00:00:00 2001
From: Vlad Pelin <vlad.pelin@nxp.com>
Date: Mon, 14 Nov 2022 14:26:55 +0200
Subject: [PATCH 1/7] pkcs: fix ec key import

after removing the DER encoding/compression header
from the EC key material, the appropriate field
needs to be set in the hse key import service
descriptor

Issue: ALB-9530
Upstream-Status: Pending 

Signed-off-by: Vlad Pelin <vlad.pelin@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 libpkcs/pkcs11_object.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libpkcs/pkcs11_object.c b/libpkcs/pkcs11_object.c
index c85742e..a8a768f 100644
--- a/libpkcs/pkcs11_object.c
+++ b/libpkcs/pkcs11_object.c
@@ -11,7 +11,6 @@
 #include "simclist.h"
 #include "hse-internal.h"
 
-
 static uint16_t getkeybitlen(hseEccCurveId_t eccCurveId)
 {
 	switch(eccCurveId) {
@@ -277,7 +276,7 @@ CK_DEFINE_FUNCTION(CK_RV, C_CreateObject)(
 			ec_point = getattr_pval(pTemplate, CKA_EC_POINT, ulCount);
 			ec_point = (uint8_t *)ec_point + 3;
 			memcpy(pkey0, ec_point, pkey0_len);
-			
+
 			ecc_oid = getattr_pval(pTemplate, CKA_EC_PARAMS, ulCount);
 			if (ecc_oid == NULL) {
 				rc = CKR_ARGUMENTS_BAD;
@@ -296,6 +295,7 @@ CK_DEFINE_FUNCTION(CK_RV, C_CreateObject)(
 			import_key_req->keyLen[0] = pkey0_len;
 			import_key_req->keyLen[1] = 0u;
 			import_key_req->keyLen[2] = 0u;
+			import_key_req->keyFormat.eccKeyFormat = HSE_KEY_FORMAT_ECC_PUB_RAW;
 
 			if (key->key_class == CKO_PRIVATE_KEY) {
 
-- 
2.17.1

